### ISSUE TITLE
Seg Fault when try to do air_loop.remove or plant_loop.remove after doing building.remove

### ISSUE THREAD

**Issue by DavidGoldwasser** [NREL/OpenStudio#2416]
I'm trying to remove non-resource objects out of model to build up new geometry/zones, so I used model.getBuilding.remove. I was then catching objects it doesn't remove like air loops and plant loops. Both of those hit seg fault if I try to remove from the model. I can reverse my order for now, but seems like this shouldn't happen.

`[BOOST_ASSERT] <2> Assertion result failed on line 307 of openstudio::model::HVACComponent openstudio::model::detail::PortList_Impl::hvacComponent() const in file /Users/kbenne/Development/OpenStudio/openstudiocore/src/model/PortList.cpp.
/Users/dgoldwas/Documents/GitHub/NREL/OpenStudio-measures/NREL working measures/create_bar_from_model/measure.rb:214: [BUG] Segmentation fault
ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin15]

-- Crash Report log information --------------------------------------------
   See Crash Report log file under the one of following:
     * ~/Library/Logs/CrashReporter
     * /Library/Logs/CrashReporter
     * ~/Library/Logs/DiagnosticReports
     * /Library/Logs/DiagnosticReports
   the more detail of.`

**Comment by jmarrec** [comment]
This looks similar to #2449 which deals with clone. Looks like building doesn't "know" which objects it's supposed to delete or clone.

**Comment by jmarrec** [comment]
I cannot reproduce this, and it seems it was probably fixed via #3534

```ruby
m = OpenStudio::Model::exampleModel
a = m.getAirLoopHVACs[0]
m.getBuilding.remove
a.remove
```

No problem
